src/components/avatar-rectangle/__tests__/avatar-rectangle.test.tsx

```diff
@@ -0,0 +1,40 @@
+import { render } from '@testing-library/react'
+import { AvatarRectangle } from '..'
+
+describe('AvatarRectangle', () => {
+  it('renders correctly with residential variant and medium size', () => {
+    const { asFragment } = render(
+      <AvatarRectangle variant="residential" size="medium" src="test-image.jpg" alt="Test Image" />,
+    )
+    expect(asFragment()).toMatchSnapshot()
+  })
+
+  it('renders correctly with residential variant and small size without src', () => {
+    const { asFragment } = render(<AvatarRectangle variant="residential" size="small" />)
+    expect(asFragment()).toMatchSnapshot()
+  })
+
+  it('renders correctly with commercial variant and small size', () => {
+    const { asFragment } = render(
+      <AvatarRectangle variant="commercial" size="small" src="test-image.jpg" alt="Test Image" />,
+    )
+    expect(asFragment()).toMatchSnapshot()
+  })
+
+  it('renders correctly with commercial variant and medium size without src', () => {
+    const { asFragment } = render(<AvatarRectangle variant="commercial" size="medium" />)
+    expect(asFragment()).toMatchSnapshot()
+  })
+
+  it('renders correctly when image fails to load', () => {
+    const { asFragment } = render(
+      <AvatarRectangle variant="commercial" size="medium" src="invalid-image.jpg" alt="Test Image" />,
+    )
+    // Simulate image error
+    const img = document.querySelector('img')
+    if (img) {
+      img.dispatchEvent(new Event('error'))
+    }
```

Comment:
I still don't understand sorry. You are providing a URL for the image that will fail to be retrieved (`invalid-image.jpg`), so the fallback to the placeholder should happen automatically. If you have to manually trigger an error event on the `img`, it makes me doubt that the fallback handling is not actually working as intended...

Reply:
its actually because in the test env it doesn't be have like that, instead it unexpectedly assume as the image is valid

![image](https://github.com/user-attachments/assets/e3efe45b-104e-4e1c-b815-7156e4d75331)

Reply:
Ahh, rightio. Can you add a comment explaining this in the test then? 🙏

Reply:
Ok, no problem